### PR TITLE
Make name optional on contextual menu, update examples on how to place dividers

### PR DESCRIPTION
**Pull request by christiango**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or **enhancement**

- [ ] Documentation update

#### Description of changes
The existing ContextualMenu examples place dividers in the menu like this:
```typescript
{
  key: 'divider_3',
  name: '-',
}
```
This doesn't seem to be a best practice, since it relies on the trick that using '-' for the name results in a divider being rendered.

Instead, the recommended approach should be to render menu items like this:
```typescript
{
  key: 'divider_3',
  itemType: ContextualMenuItemType.Divider
}
```
In order to enable this, we should make the name property optional on IContextualMenuItem.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Basic.Example.tsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class ContextualMenuBasicExample extends React.Component<any, any> {
                 },
                 {
                   key: 'divider_1',
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Basic.Example.tsx]: would it make sense that the key is generated by the component when the type is divider?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Basic.Example.tsx]: I think `key` could really always be generated if not provided:

```
key = key | name | index
```

something like that?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Basic.Example.tsx]: But there is a bit of a bad side effect, which is if you don't provide key, then anything with state may not work as expected.

Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Basic.Example.tsx]: Personally, I'd rather we require the key. I don't want us to end up in a situation where we are using the index for a key inside the component.

**Pull request merged**